diff --git a/Makefile b/Makefile
index 322dc67..910226a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,17 @@
+WORKDIR=$(PWD)
+BIN=$(WORKDIR)/bin
+
+UNAME := $(shell uname -s)
+
+ifeq ($(UNAME), Linux)
+# do something Linux-y
+machine=Linux
+endif
+ifeq ($(UNAME), Darwin)
+# do something Solaris-y
+machine=OSX
+endif
+
 FS = src/FSWeighting
 FSE = FSWeight
 CA = src/CAw
@@ -8,8 +22,8 @@ FSCE = FScoring
 all:	compile fscore
 	
 compile:
-	g++ $(FS).cpp -o $(FSE)
-	g++ $(CA).cpp -o $(CAE)
+	g++ $(FS).cpp -o $(BIN)/$(machine)/$(FSE)
+	g++ $(CA).cpp -o $(BIN)/$(machine)/$(CAE)
 
 fscore:
 	(cd scoring; make )
